lib/libvmod_blob/vmod_blob.h

```diff
@@ -107,34 +104,29 @@ ssize_t encode_f(const enum encoding enc, const enum case_e kase,
  *    a static constant empty BLOB
  * otherwise, the number of bytes written
  */
-typedef ssize_t decode_f(const enum encoding dec, char *restrict const buf,
-    const size_t buflen, const ssize_t inlen, VCL_STRANDS strings);
+typedef ssize_t decode_f(BLOB_CODEC, char *buf, size_t buflen, ssize_t inlen,
+    VCL_STRANDS strings);
+
+struct vmod_blob_codec {
```

Comment:
"codec" is indeed a better name.

Consider adding a magic number to struct vmod_blob_codec, then it could be checked with CHECK_OBJ (currently just with AN).